src/Factory.php

```diff
@@ -68,8 +68,7 @@ public function __call(string $name, array $arguments)
     /**
      * @param array|callable $attributes
      *
-     * @return Proxy|object
-     *
+     * @return Proxy<TObject>&TObject
```

Comment:
No need for `&TObject`, as the `Proxy` mixin is already taking care of this?

Reply:
You are correct. However, PHPStorm do not understand that. 

So we add the `@return` to make sure PHPStorm is auto completing correctly, then we do `@psalm-return` with the "correct" return type so psalm does not complain.